front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/ElectricalProfilesLegend.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface Props {
+  isActive: boolean;
+  setIsActive: Function;
+}
+
+interface Profile {
+  mode: string;
+  color: string[];
+  isStriped: boolean;
+}
+
+export const ElectricalProfilesLegend = ({ isActive, setIsActive }: Props) => {
+  const { t } = useTranslation('simulation');
+
+  const legend: Profile[] = [
```

Comment:
Please set these consts outside the context of this functionnal prop

Reply:
done